third_party/libffi.BUILD

```diff
@@ -0,0 +1,35 @@
+# Copyright 2019 Google LLC
```

Comment:
What is this? I think it is a bit strange that this file is at the root of the `third_party` folder. 
Also, could not it be a dependency in the workspace?

Reply:
Does this file have to be in the root of `third_party`, instead of for instance being in a `third_party/libffi` folder? Is it possible to remove this file and move its content to `third_party/BUILD`, or is this particular file name important? The license header is different from the other files. Is it because the content of this file is copied from somewhere else?

Cannot we directly use "https://github.com/kythe/kythe/blob/a367ca6f/third_party/libffi.BUILD" in `http_archive` for `org_sourceware_libffi` instead of repeating the bazel targets in this file here?

Reply:
So we can't import the official release of `libffi` (from https://github.com/libffi/libffi/releases/download/v3.3/libffi-3.3.tar.gz) using `http_archive` directly followed by `load` like you have for the other `http_archive` rules in this repo because libffi's release does not contain a build file. So to use that official release, we need to add our own build file (here in `//third_party:libffi.BUILD` and connect this to the rule with the `build_file` param).

It looks like the `kythe` repository you linked with a `libffi.BUILD` is for a different project, so we probably should not use this for our libffi dependency? I think to use an `http_archive` followed by a `load` you need to have both an archive and the BUILD, but that kythe project does not have a n archive released.

Reply:
Ah, ok. I did not notice that `third_party/libffi.BUILD` is copied from a repo other than https://github.com/libffi.

I am still not convinced why the license is different. Can you update the license and keep the note that the targets are copied from somewhere else, also add a comment explaining why this is needed?

Reply:
It's removed now.